Python/ceval.c

```diff
@@ -5022,12 +5022,23 @@ import_from(PyObject *v, PyObject *name)
     return x;
  error:
     pkgpath = PyModule_GetFilenameObject(v);
+    if (pkgname == NULL) {
+        pkgname_or_unknown = PyUnicode_FromString("<unknown module name>");
+    } else {
+        pkgname_or_unknown = pkgname;
+    }
```

Comment:
> If you want to be pedanditc, it would be useful to check for that error condition and do something more reasonable in that case

I'm unsure what "more reasonable" is,  I check and failed with the original error message in that case, but I don't see how if allocating `PyUnicode_FromString("<unknown module name>")` fails we can do something sensible.